quinn-proto/src/frame.rs

```diff
@@ -826,6 +826,29 @@ impl Datagram {
     }
 }
 
+/// Indicates whether a frame of type `F` needs to be transmitted
+///
+/// This type is around `bool` which uses the `#[must_use]` attribute in order
+/// to prevent accidental loss of the frame transmission requirement.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+#[must_use = "A frame might need to be enqueued"]
+pub struct WantTransmit {
+    want_transmit: bool,
+}
+
+impl WantTransmit {
+    /// Creates a new `WantTransmit` instance, which indicates whether a frame
+    /// of type `F` should be transmitted
+    pub fn new(want_transmit: bool) -> Self {
+        Self { want_transmit }
+    }
+
+    /// Returns whether a frame ot type `F` should be transmitted
+    pub fn want_transmit(self) -> bool {
+        self.want_transmit
+    }
+}
```

Comment:
These methods don't feel like they carrry their weight at this point (perhaps as an effect of losing the type argument?).

Reply:
Maybe a one-public-element tuple struct? It's not like this can conceivably grow any internal logic in the future.

Or maybe an `enum`? That's a little more idiomatic, I think, but I'm not sure what good variant names would be.

Reply:
If you don't have them, you end up with checks which look like `if transmit_max_data.0 {`

I'm personally not a fan of those `.0` accesses. But I agree the duplication between method and type names isn't elegant. If someone has ideas around better names, I'm all open for it. It could be as simple as `fn yes(&self)`, but that looks weird too.

Reply:
Maybe `if max_data.transmit()`?

Reply:
That looks good. But it's kind of debatable whether to name the field in `ReadResult` `max_data`, since it doesn't contain the `max_data` value. If we get to `result.should_transmit_max_data.should_transmit()` we have the doubling again.`  `result.transmit_max_data.transmit()` is kind of the same

Reply:
I think `result.max_data.should_transmit()` reads exactly right -- it doesn't seem like a big concern that the `bool`-like value of `result.max_data` is not a `max_data` value, since that should be obvious from the type system.